back/engines/commercial/analytics/spec/factories/fact_visits.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :fact_visit, class: 'Analytics::FactVisit' do
+    transient do
+      date_dimension { Analytics::DimensionDate.first || create(:dimension_date_sept) }
+      referrer_type_dimension { Analytics::DimensionReferrerType.first || create(:dimension_referrer_type_website) }
```

Comment:
I would not make the defaults depend on the DB order, which is not reliable.
```suggestion
      date_dimension { create(:dimension_date_sept) }
      referrer_type_dimension { create(:dimension_referrer_type_website) }
```

Reply:
I did this because I wanted to use a pre-existing dimension when creating multiple visits. When I tried it the way you are suggesting it failed due to duplicate key errors.

Reply:
What key? Shall we pair on this? It should be pretty straightforward to solve.